lib/puppet/type/ios_radius_server_group.rb

```diff
@@ -0,0 +1,27 @@
+require 'puppet/resource_api'
+
+Puppet::ResourceApi.register_type(
+  name: 'ios_radius_server_group',
+  docs: 'Configure a RADIUS server group',
+  features: ['canonicalize', 'simple_get_filter'] + (Puppet::Util::NetworkDevice.current.nil? ? [] : ['remote_resource']),
```

Comment:
The `Puppet::Util::NetworkDevice.current.nil?` check is redundant and ` ['canonicalize', 'simple_get_filter', 'remote_resource']` is sufficient. 

`canonicalize` and `simple_get_filter` in the current implementation are not made use off - in netdev types we have to have them as a feature even without valid implementation, as this will be an IOS specific type we need to be stricter in listing what `features` are available, and implementing them when possible.

`canonicalize` in this sense might be useful for sorting servers/private_servers as in other places the device typically returns them in alphabetical order - at this point I have not verified if this applies to this resource.

Reply:
Yes with in the `instances_from_cli` it will be best to do a sort on the `servers` and `private_servers`

And then in `canonicalize` do something like:

```
resources.each do |resource|
  resource[:servers] = resource[:servers].sort if resource[:servers]
  resource[:private_servers] = resource[:private_servers].sort if resource[:private_servers]
end
resources
```

As currently it fails as the device seems to return last in first out:

```
Found faulty change event 'Notice: /Stage[main]/Main/Ios_radius_server_group[bill]/private_servers: private_servers changed ['5.6.7.8', '1.2.3.4'] to ['1.2.3.4', '5.6.7.8'] (corrective)' at
```

Which makes it currently not idempotent.

Reply:
Have removed 'remote_resource' and put a sort in canocalise

Reply:
Are you ok with the code as it is now?